spec/chromium-spec.js

```diff
@@ -371,16 +371,38 @@ describe('chromium feature', () => {
   })
 
   describe('storage', () => {
-    describe('DOM storage quota override', () => {
+    describe('DOM storage quota increase', () => {
       ['localStorage', 'sessionStorage'].forEach((storageName) => {
-        it(`allows saving at least 50MiB in ${storageName}`, () => {
-          const storage = window[storageName]
-          const testKeyName = '_electronDOMStorageQuotaOverrideTest'
-          // 25 * 2^20 UTF-16 characters will require 50MiB
-          const arraySize = 25 * Math.pow(2, 20)
-          storage[testKeyName] = new Array(arraySize).fill('X').join('')
-          expect(storage[testKeyName]).to.have.lengthOf(arraySize)
-          delete storage[testKeyName]
+        const storage = window[storageName]
+        it(`allows saving at least 40MiB in ${storageName}`, (done) => {
+          // Although JavaScript strings use UTF-16, the underlying
+          // storage provider may encode strings differently, muddling the
+          // translation between character and byte counts. However,
+          // a string of 40 * 2^20 characters will require at least 40MiB
+          // and presumably no more than 80MiB, a size guaranteed to
+          // to exceed the original 10MiB quota yet stay within the
+          // new 100MiB quota.
+          // Note that both the key name and value affect the total size.
+          const testKeyName = '_electronDOMStorageQuotaIncreasedTest'
+          const length = 40 * Math.pow(2, 20) - testKeyName.length
+          storage.setItem(testKeyName, 'X'.repeat(length))
+          // Wait at least one turn of the event loop to help avoid false positives
```

Comment:
What false positives might these be? Ideally we could simplify this test to not require a tick.

Reply:
Good question. This might not be applicable any more, but when we were circumventing the restriction prior to this PR, a write followed immediately by a read would "appear to work" (no exceptions throw & reading shows the expected value). Now that we're letting chromium behave more like its usual self (i.e. throwing an exception when a write would exceed the allowed limit) these "false positives" should've be possible any more. Would you like me to remove the `setTimeout`?

Reply:
ah, okay. In that case it seems fine to keep the extra tick, it will let us know if things start behaving weirdly again. It'd be great if you could add that context into this comment, though, as it may help the next person to work with this code!

Reply:
Updated -- hope that's useful (I'm often wary of comments as they can accidentally be misleading)